horovod/spark/common/util.py

```diff
@@ -0,0 +1,518 @@
+                size = len(col_data.indices)
+                padding_zeros = 2 * (metadata[col]['max_size'] - len(col_data.indices))
+
+                fields[col] = np.concatenate(
+                    (np.array([size]), col_data.indices, col_data.values,
+                     np.zeros(padding_zeros))).tolist()
+
+        return Row(**fields)
+
+    return to_petastorm
+
+
+def get_simple_meta_from_parquet(store, label_columns, feature_columns, sample_weight_col):
+    train_data_path = store.get_train_data_path()
+    validation_data_path = store.get_val_data_path()
+
+    if not store.exists(train_data_path):
+        raise ValueError("{} path does not exist on the store".format(train_data_path))
+
+    train_data = pq.ParquetDataset(train_data_path, filesystem=store.get_filesystem())
+    schema = train_data.schema.to_arrow_schema()
+
+    train_rows = 0
+    total_byte_size = 0
+    for piece in train_data.pieces:
+        metadata = piece.get_metadata()
+        train_rows += metadata.num_rows
+
+        pfile = piece.open()
+        file_metadata = pfile.metadata
+        for row_group_index in range(file_metadata.num_row_groups):
+            row_group = file_metadata.row_group(row_group_index)
+            total_byte_size += row_group.total_byte_size
+
+    if train_rows == 0:
+        raise ValueError('No rows found in training dataset: {}'.format(train_data_path))
+
+    if total_byte_size == 0:
+        raise ValueError('No data found in training dataset: {}'.format(train_data_path))
+
+    if train_rows > total_byte_size:
+        raise ValueError('Found {} bytes in {} rows.  Training dataset may be corrupted.'
+                         .format(total_byte_size, train_rows))
+
+    val_rows = 0
+    if store.exists(validation_data_path):
+        val_data = pq.ParquetDataset(validation_data_path, filesystem=store.get_filesystem())
+        for piece in val_data.pieces:
+            metadata = piece.get_metadata()
+            val_rows += metadata.num_rows
+
+    schema_cols = feature_columns + label_columns
+    if sample_weight_col:
+        schema_cols.append(sample_weight_col)
+
+    metadata = {}
+    for col in schema_cols:
+        col_schema = schema.field_by_name(col)
+        col_info = {
+            'spark_data_type': pyarrow_to_spark_data_type(col_schema.type),
+            'is_sparse_vector_only': False,
+            'shape': None,  # Only used by SparseVector columns
+            'intermediate_format': constants.NOCHANGE,
+            'max_size': None  # Only used by SparseVector columns
+        }
+        metadata[col] = col_info
+
+    avg_row_size = total_byte_size / train_rows
+    return train_rows, val_rows, metadata, avg_row_size
+
+
+def prepare_data(num_processes, store, df, label_columns, feature_columns,
+                 validation_col=None, validation_split=0.0, sample_weight_col=None,
+                 partitions_per_process=10, verbose=False):
+    if validation_split and validation_col:
```

Comment:
Then you could have a single argument `validation` that can be either a String (interpreted as a validation column) or a float (interpreted as a validation split).

Reply:
`validation_split` must be `[0.0, 1)`, is this checked where user code calls into horovod (if this is not that place)?

Reply:
I like the idea of merging `validation_split` and `validation_col`.  @abditag2 what do you think?

I'll also add the check on the validation split.  Thanks for the suggestion.

Reply:
Done.